sources/api/netdog/src/interface_name.rs

```diff
@@ -10,16 +10,16 @@ use std::ops::Deref;
 
 /// InterfaceName can only be created from a string that contains a valid network interface name.
 /// Validation is handled in the `TryFrom` implementation below.
-#[derive(Debug, Eq, PartialEq, Hash, Deserialize)]
-#[serde(try_from = "&str")]
+#[derive(Clone, Debug, Eq, PartialEq, Hash, Deserialize)]
+#[serde(try_from = "String")]
```

Comment:
Since we now deserialize the majority of the network config to a `toml::Value` first (to handle the versions), this is stored as a `String` inside of the `toml::Value`, hence the change from `&str` to `String`.